### DISCUSSION TITLE
loops through table_0 ... table_999?

### DISCUSSION THREAD

**Discussion by vinh-nguyen-phd** [duckdb/duckdb#3009] [Q&A] [answered]
Hi everyone, 

I have a table with 15 columns and several hundred mm rows which I split into 1000 parquet files. I need to merge this with a right table with 100 thousand rows. Is there a way to loop through the 1000 parquet files and do the same merge with the right table? Below is the sample code that I want to do with the loop. Thank you.

con.execute("""
   SELECT DISTINCT a.*, b*
         FROM PARQUET_SCAN(table_0.parquet) as a 
         INNER JOIN right_table as b
         ON a.id = b.id          
         ;
""")
...............
...............
con.execute("""
   SELECT DISTINCT a.*, b*
         FROM PARQUET_SCAN(table_999.parquet) as a 
         INNER JOIN right_table as b
         ON a.id = b.id          
         ;
""")

**Comment by hannes** [comment, marked as answer]
You're already in Python so you could loop there:
```Python
for i in range(1000):
  con.execute("""
  SELECT DISTINCT a, b
  FROM PARQUET_SCAN('table_%d.parquet') as a
  INNER JOIN right_table as b
  ON a.id = b.id
  """ % i)
```

Or you can use a wildcard to read all parquet files in one query and then join, e.g.

```SQL
SELECT DISTINCT a., b
FROM PARQUET_SCAN('table_*.parquet') as a
INNER JOIN right_table as b
ON a.id = b.id
```


Overall not so sure what you're trying to achieve here.

**Reply by Alex-Monahan** [reply]
You beat me to it Hannes! I totally agree! :-)

**Reply by hannes** [reply]
Night shift at it again :)

**Reply by vinh-nguyen-phd** [reply]
Hi, I think I find the solution below. I will let it run and update this thread in the morning. Thanks.

for i in range(0,1000,1):
        

con.execute("""
         COPY (SELECT DISTINCT a.something , b.something_else
         FROM PARQUET_SCAN(table_{}.parquet) as a
         INNER JOIN right_table as b
         ON a.id = b.id)
         TO 'merge_{}.parquet' (FORMAT 'PARQUET')
         ;
         """. format(i,i))

**Reply by vinh-nguyen-phd** [reply]
So, the code that I posted above works in my case. Thanks again for all the help.

**Comment by Alex-Monahan** [comment]
Thanks for trying out DuckDB!

If you would like to treat many parquet files as if they were one table, I would recommend using the glob pattern matching syntax! Using a * is a wildcard.

For example, you could do:
```python
con.execute("""
SELECT DISTINCT a.*, b.*
FROM PARQUET_SCAN(*.parquet) as a
INNER JOIN right_table as b
ON a.id = b.id
;
""")
```
Here are some [additional parquet examples!](https://duckdb.org/docs/data/parquet)

**Comment by vinh-nguyen-phd** [comment]
Thanks so much for the helpful responses. I will try out all methods.
